### PR TITLE
treewide: update darwin sdk hashes

### LINKED ISSUE #201348: Darwin opensource packages are no longer available
###### Project name
This is in the file: `pkgs/os-specific/darwin/apple-source-releases/macos-11.0.1.nix`

###### Notify maintainers

maintainers: ???

###### Note for maintainers

`bootstrap_cmds`, for example, is given at version 121, but fails to download from every machine I've tried:
```
wget http://www.opensource.apple.com/tarballs/bootstrap_cmds/bootstrap_cmds-121.tar.gz
```

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Closes #201348

(But I'm not sure, since the hashes were updated manually...)

Feel free to edit the commit, especially if some of the hashes were wrong.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build stdenv

File: pkgs/os-specific/darwin/apple-source-releases/adv_cmds/boot.nix
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/apple-source-releases/adv_cmds/boot.nix:13]: Pelase use fetchFromGitHub

File: pkgs/tools/filesystems/hfsprogs/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/hfsprogs/default.nix:12]: Same here

File: pkgs/tools/system/pdisk/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/system/pdisk/default.nix:16]: and here

File: pkgs/os-specific/darwin/apple-source-releases/default.nix
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/apple-source-releases/default.nix:262]: and here

File: pkgs/os-specific/darwin/apple-source-releases/default.nix
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/apple-source-releases/default.nix:149]: and probably here, too

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
FYI due to cache misses ofborg will unlikely build anything for x86_64-darwin.

File: pkgs/os-specific/darwin/apple-source-releases/default.nix
Comment (wegank) [reply pkgs/os-specific/darwin/apple-source-releases/default.nix:149]: I've fixed the others, but this one seems tricky.

**Comment by Artturin** [comment]
> > Is this blocked on anything? Are you trying to update the `fetch` to `fetchFromGithub`?
> 
> It was blocked by the changes in [community/community#46034](https://github.com/orgs/community/discussions/46034), which seems to be reverted now.

fetchFromGitHub and fetchzip hash the output of the archive not the archive itself so the change doesn't affect us

**Comment by wegank** [comment]
We're using `stdenv.fetchurlBoot` and `fetchurl` in this PR, unfortunately...

**Comment by winterqt** [comment]
It seems that Apple changed the locations of some headers, which breaks IOKit; see https://github.com/NixOS/nixpkgs/pull/215401#issuecomment-1427049425.

Please test things like this in the future, when possible (especially if OfBorg can't). If compute power is stopping you, see https://github.com/winterqt/darwin-build-box.

**Comment by winterqt** [comment]
Also: @Artturin, please don't disregard comments from a committer and go ahead and merge something, as they were obviously said for a reason. Thanks.

**Comment by Artturin** [comment]
> Also: @Artturin, please don't disregard comments from a committer and go ahead and merge something, as they were obviously said for a reason. Thanks.

I don't know what comment you're referring to

**Comment by winterqt** [comment]
> > Also: @Artturin, please don't disregard comments from a committer and go ahead and merge something, as they were obviously said for a reason. Thanks.
> 
> I don't know what comment you're referring to

Ah, apologies.

I meant https://github.com/NixOS/nixpkgs/pull/210449#issuecomment-1418688838, but I misunderstood and thought it was still blocked on it at the time.

Either way, your comment (https://github.com/NixOS/nixpkgs/pull/210449#issuecomment-1419045070) isn't true because of https://github.com/NixOS/nixpkgs/pull/210449#issuecomment-1419047794.

Sorry, though -- my apologies for misreading the dialogue here.

**Pull request merged**
